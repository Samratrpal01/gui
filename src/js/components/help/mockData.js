export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {},
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '01-01-2003'
  }
};
