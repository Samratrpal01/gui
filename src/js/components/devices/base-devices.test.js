import React from 'react';

import { defaultState, undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';
import {
  AcceptedEmptyState,
  DeviceCreationTime,
  DeviceExpansion,
  DeviceStatusHeading,
  PendingEmptyState,
  PreauthorizedEmptyState,
  RejectedEmptyState,
  RelativeDeviceTime
} from './base-devices';

describe('smaller components', () => {
  [
    AcceptedEmptyState,
    DeviceCreationTime,
    DeviceExpansion,
    DeviceStatusHeading,
    PendingEmptyState,
    PreauthorizedEmptyState,
    RejectedEmptyState,
    RelativeDeviceTime
  ].forEach(Component => {
    it(`renders ${Component.displayName || Component.name} correctly`, () => {
      const { baseElement } = render(
        <Component filters={[]} highlightHelp={true} limitMaxed={true} onClick={jest.fn} allCount={10} device={defaultState.devices.byId.a1} />
      );
      const view = baseElement.firstChild.firstChild;
      expect(view).toMatchSnapshot();
      expect(view).toEqual(expect.not.stringMatching(undefineds));
    });
  });
});
