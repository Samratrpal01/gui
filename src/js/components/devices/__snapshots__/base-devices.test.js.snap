// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smaller components renders AcceptedEmptyState correctly 1`] = `
<div
  class="dashboard-placeholder"
>
  <p>
    No devices found
  </p>
</div>
`;

exports[`smaller components renders DeviceCreationTime correctly 1`] = `-`;

exports[`smaller components renders DeviceExpansion correctly 1`] = `
.emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

<div
  class="bold flexbox center-aligned link-color margin-right-small uppercased"
  style="white-space: nowrap;"
>
  view details 
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-0"
    data-testid="ArrowRightAltIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.01 11H4v2h12.01v3L20 12l-3.99-4z"
    />
  </svg>
</div>
`;

exports[`smaller components renders DeviceStatusHeading correctly 1`] = `<div />`;

exports[`smaller components renders PendingEmptyState correctly 1`] = `
<div
  class="dashboard-placeholder"
>
  <p>
    There are no devices pending authorization
  </p>
  <p>
    Visit the 
    <a
      href="/help/get-started"
    >
      Help section
    </a>
     to learn how to connect devices to the Mender server.
  </p>
</div>
`;

exports[`smaller components renders PreauthorizedEmptyState correctly 1`] = `
<div
  class="dashboard-placeholder"
>
  <p>
    There are no preauthorized devices.
  </p>
  <p>
    Preauthorize devices
     so that when they come online, they will connect to the server immediately
  </p>
  <img
    alt="preauthorize"
    src="preauthorize.png"
  />
</div>
`;

exports[`smaller components renders RejectedEmptyState correctly 1`] = `
<div
  class="dashboard-placeholder"
>
  <p>
    There are no rejected devices
  </p>
</div>
`;

exports[`smaller components renders RelativeDeviceTime correctly 1`] = `
<span
  aria-label=""
  class=""
  data-mui-internal-clone-element="true"
>
  <div>
    -
  </div>
</span>
`;
